src/subcommand/fix.rs

```diff
@@ -0,0 +1,484 @@
+                    }
+                }
+            }
+        }
+    }
+
+    #[tracing::instrument]
+    fn fossa_get_explanation(
+        description: &str,
+        specific_error_message: &str,
+        url: &str,
+        example_command: &str,
+        err: reqwest::Error,
+    ) -> String {
+        formatdoc!(
+            "{}
+
+            {}
+
+            The URL we attempted to connect to was {}. Please make sure you can make a request to that URL. For example, try this curl command:
+
+            {}
+
+            Full error message: {}
+            ",
+            description.red(),
+            specific_error_message,
+            url,
+            example_command.green(),
+            err
+        )
+    }
+}
+
+/// The primary entrypoint for the fix command.
+#[tracing::instrument(skip_all, fields(subcommand = "fix"))]
+pub async fn main(config: &Config) -> Result<(), Report<Error>> {
+    let integration_errors = check_integrations(config).await;
+    let fossa_connection_errors = check_fossa_connection(config).await;
+    print_errors(
+        "\nErrors found while checking integrations",
+        integration_errors,
+    );
+    print_errors(
+        "\nErrors found while checking connection to FOSSA",
+        fossa_connection_errors,
+    );
+    Ok(())
+}
+
+// If there are errors, returns a string containing all of the error messages for a section.
+// Sections are things like "checking integrations" or "checking fossa connection"
+// If there are no errors, it returns None.
+#[tracing::instrument]
+fn print_errors(msg: &str, errors: Vec<Error>) {
+    if !errors.is_empty() {
+        println!("{}\n", msg.bold().red());
+        for err in errors {
+            println!("{}", err.fix_explanation());
+        }
+    }
+}
+
+/// Check that we can connect to the integrations
+/// This is currently done by running `git ls-remote <remote>` using the authentication
+/// info from the transport.
+#[tracing::instrument(skip(config))]
+async fn check_integrations(config: &Config) -> Vec<Error> {
+    let title = "\nDiagnosing connections to configured repositories\n"
+        .bold()
+        .blue()
+        .to_string();
+    println!("{}", title);
+    let integrations = config.integrations();
+    let mut errors = Vec::new();
+    for integration in integrations.iter() {
+        match check_integration(integration).await {
+            Ok(()) => {
+                println!("✅ {}", integration.remote())
+            }
+            Err(err) => {
+                println!("❌ {}", integration.remote());
+                errors.push(err);
+            }
+        }
+    }
+    errors
+}
+
+#[tracing::instrument]
+async fn check_integration(integration: &Integration) -> Result<(), Error> {
+    let Protocol::Git(transport) = integration.protocol();
+    repository::ls_remote(transport)
+        .await
+        .or_else(|err| Error::integration_error(integration.remote(), transport, err).wrap_err())?;
+    Ok(())
+}
```

Comment:
This is a function we'd actually want to debug, so I'd keep tracing on this kind of function.

Basically rather than throwing `#[tracing::instrument]` on everything, just put it on things that are potentially something we'd want to debug or otherwise instrument (e.g. timing info). Formatting or type conversion functions I think not so much, but things like running external commands or fallible functions generally are definite shoe-ins.